addons/web/static/src/search/utils/dates.js

```diff
@@ -147,17 +146,20 @@ export function constructDateDomain(
     if (comparisonOptionId) {
         [plusParam, selectedOptions] = getComparisonParams(
             referenceMoment,
+            searchItem,
             selectedOptionIds,
             comparisonOptionId
         );
     } else {
-        selectedOptions = getSelectedOptions(referenceMoment, selectedOptionIds);
+        selectedOptions = getSelectedOptions(referenceMoment, searchItem, selectedOptionIds);
     }
     const yearOptions = selectedOptions.year;
```

Comment:
if ("domain" in selectedOptions) {
const { domain, description } = selectedOptions.domain[0];
return { domain: customOption.domain, description: customOption.description };
}
? 
If I understand the spec, this should be simpler